core/src/main/scala/cats/instances/list.scala

```diff
@@ -19,6 +20,14 @@ trait ListInstances extends cats.kernel.instances.ListInstances {
 
       def combineK[A](x: List[A], y: List[A]): List[A] = x ::: y
 
+      override def combineAllOptionK[A](as: IterableOnce[List[A]]): Option[List[A]] = {
+        val iter = as.iterator
+        if (iter.isEmpty) None else Some(iter.flatMap(_.iterator).toList)
+      }
+
+      override def combineAllK[A](as: IterableOnce[List[A]]): List[A] =
+        as.iterator.flatMap(_.iterator).toList
```

Comment:
this duplication is unfortunate IMO. I think in MonoidK we should be able to only override one.

In library code I think it is worth the verbosity for the possibility of slightly better performance:
```scala
val bldr = List.newBuilder[A]
val it = as.iterator
while(it.hasNext) {
  bldr ++= it.next()
}
bldr.result()
```